packages/data-context/src/sources/GitDataSource.ts

```diff
@@ -217,21 +207,15 @@ export class GitDataSource {
     }
 
     try {
-      const [stdout, statusResult] = await Promise.all([
-        os.platform() === 'win32'
-          ? this.#getInfoWindows(absolutePaths)
-          : this.#getInfoPosix(absolutePaths),
-        this.#git.status(),
-      ])
-
-      debug('stdout %s', stdout)
+      const statusResult = await this.#git.status()
 
       const changed: string[] = []
 
       // Go through each file, updating our gitInfo cache and detecting which
       // entries have changed, to notify the UI
-      for (const [i, file] of absolutePaths.entries()) {
+      for (const file of absolutePaths) {
```

Comment:
So I forgot about this in the pairing session earlier, but part of the reason we had it the way we did previously was because simpleGit spawns a child process per command, which gets very expensive when the number of specs grows. The way we have it currently batches it all into a single command to save on child process overhead.

I haven't looked closely at their API to determine if there's a way to batch them, they do have some notes about it here but it's a little vague: https://github.com/steveukx/git-js#concurrent--parallel-requests

If we can figure out why it's broken with the current approach and how to fix, that might be better - if we determine it's worth it to stick to simple git, we prob want to change this into a `Promise.all(absolutePaths.map(...` rather than a for loop, because the `for / await` will block and run sequentially rather than concurrently.